src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -473,15 +472,164 @@ def filter_devices(self, rook_pods: KubernetesResource, drive_group: DriveGroupS
                             count += 1
         return device_list
 
+class DefaultRemover():
+    def __init__(
+        self,
+        coreV1_api: 'client.CoreV1Api', 
+        batchV1_api: 'client.BatchV1Api', 
+        appsV1_api: 'client.AppsV1Api', 
+        osd_ids: List[str], 
+        replace_flag: bool, 
+        force_flag: bool, 
+        mon_command: Callable, 
+        patch: Callable, 
+        rook_env: 'RookEnv',
+        pods: KubernetesResource, 
+        inventory: Dict[str, List[Device]]
+    ):
+        self.batchV1_api = batchV1_api
+        self.appsV1_api = appsV1_api
+        self.coreV1_api = coreV1_api
+
+        self.osd_ids = osd_ids
+        self.replace_flag = replace_flag
+        self.force_flag = force_flag
+
+        self.mon_command = mon_command
+
+        self.patch = patch
+        self.rook_env = rook_env
+
+        self.pods = pods
```

Comment:
Here we only need OSDs, so i suggest to get the list of OSDs directly  here in the initialization, and not pass the complete list of pods:

`self.pods : KubernetesResource = KubernetesResource(self.coreV1_api.list_namespaced_pod, namespace='rook-ceph', selector='osd')`

This change will also make possible to simplify the method "remove_device_sets" to filter only by osd_ids.